prometheus/process_collector.go

```diff
@@ -33,6 +33,14 @@ type processCollector struct {
 	inBytes, outBytes *Desc
 }
 
+func init() {
+	// Debugging to show the correct implementation is used based on CGO_ENABLED=0.
```

Comment:
Do we want to merge that though?

Reply:
> Do we want to merge that though?

No.  This initial PR was just throwing a dart to show it working, and get either a "this cgo is bad, please avoid it", or "3rd party deps are bad, use cgo", or "this isn't workable at all" response to resolve the various discussions about #1590.

> Let me review deeper, is this ready to go then?

It sounds like you're agreeable to the general idea here, so give me a day or two to migrate this to the APIs that I used in the purego version for portability.  Is there a preferred format for C code, or is anything fine as long as it's reasonable?